name.abuchen.portfolio.ui/src/name/abuchen/portfolio/ui/views/earnings/EarningsViewModel.java

```diff
@@ -233,38 +239,130 @@ private void calculate()
-
-                Money dividendValue = useGrossValue ? t.getGrossValue() : t.getMonetaryAmount();
-                long value = dividendValue.with(converter.at(t.getDateTime())).getAmount();
-                if (t.getType().isDebit())
-                    value *= -1;
-
-                int index = (t.getDateTime().getYear() - startYear) * 12 + t.getDateTime().getMonthValue() - 1;
-
-                InvestmentVehicle vehicle = t.getSecurity() != null ? t.getSecurity() : account;
-                Line line = vehicle2line.computeIfAbsent(vehicle, s -> new Line(s, noOfmonths));
-                line.values[index] += value;
-                line.sum += value;
-
-                sum.values[index] += value;
-                sum.sum += value;
+                long value = 0;
+                switch(mode) {
+                    case TAXES:
+                        if (t.getType() == AccountTransaction.Type.TAXES
+                                        || t.getType() == AccountTransaction.Type.TAX_REFUND)
+                        {
+                            value = t.getMonetaryAmount().with(converter.at(t.getDateTime())).getAmount();
+                            if (t.getType().isDebit())
+                                value *= -1;
+                        }
+                        else
+                            value -= t.getUnitSum(Unit.Type.TAX).with(converter.at(t.getDateTime())).getAmount();
+                        break;
+                    case FEES:
+                        if (t.getType() == AccountTransaction.Type.FEES
+                                        || t.getType() == AccountTransaction.Type.FEES_REFUND)
+                        {
+                            value = t.getMonetaryAmount().with(converter.at(t.getDateTime())).getAmount();
+                            if (t.getType().isDebit())
+                                value *= -1;
+                        }
+
+                        else
+                            value -= t.getUnitSum(Unit.Type.FEE).with(converter.at(t.getDateTime())).getAmount();
+                        break;
+                    case ALL:
+                        if (t.getType() == AccountTransaction.Type.TAXES
+                                        || t.getType() == AccountTransaction.Type.TAX_REFUND
+                                        || t.getType() == AccountTransaction.Type.FEES
+                                        || t.getType() == AccountTransaction.Type.FEES_REFUND)
+                            value = t.getMonetaryAmount().with(converter.at(t.getDateTime())).getAmount();
+                        else
+                            value = t.getMonetaryAmount().with(converter.at(t.getDateTime())).getAmount();
+                        if (t.getType().isDebit())
+                            value *= -1;
+                        break;
+
+                    default:
+                        value = (useGrossValue ? t.getGrossValue() : t.getMonetaryAmount())
```

Comment:
Weil "Brutto/Netto" ja nur für die ersten drei Ansichten relevant ist, habe ich a) das Menü ausgeblendet wenn es nicht relevant ist und b) umbenannt in Brutto/Netto (es hießt bisher "Bruttodividende" aber es sind ja nicht mehr nur Dividenden).